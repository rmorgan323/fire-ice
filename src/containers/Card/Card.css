.card-component {
  width: 300px;
  margin: 10px;
}

.card-component h2 {
  text-align: center;
}

.card-component h3 {
  text-align: left;
}