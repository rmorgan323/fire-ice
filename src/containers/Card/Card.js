import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import './Card.css';

const Card = ({ name, founded, seats, titles, coatOfArms, ancestralWeapons, words }) => {

  return (
    <div className="card-component">
      <h2>{name}</h2>
      <h3>Founded: {founded || 'unknown'}</h3>
      <h3>Seats: {seats}</h3>
      <h3>Titles: {titles}</h3>
      <h3>Coat of Arms: {coatOfArms}</h3>
      <h3>Ancestral Weapons: {ancestralWeapons}</h3>
      <h3>Words: {words || 'unknown'}</h3>
    </div>
  )
}

export default Card;
