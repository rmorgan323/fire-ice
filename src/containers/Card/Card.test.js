import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('card tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<Card />)

    expect(renderedApp).toMatchSnapshot();
  })
})