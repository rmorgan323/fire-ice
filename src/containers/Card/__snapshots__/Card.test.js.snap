// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <h3>
          Founded: 
          unknown
</h3>,
        <h3>
          Seats: 
</h3>,
        <h3>
          Titles: 
</h3>,
        <h3>
          Coat of Arms: 
</h3>,
        <h3>
          Ancestral Weapons: 
</h3>,
        <h3>
          Words: 
          unknown
</h3>,
      ],
      "className": "card-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "unknown",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "unknown",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Titles: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Words: ",
            "unknown",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Words: ",
          "unknown",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <h3>
            Founded: 
            unknown
</h3>,
          <h3>
            Seats: 
</h3>,
          <h3>
            Titles: 
</h3>,
          <h3>
            Coat of Arms: 
</h3>,
          <h3>
            Ancestral Weapons: 
</h3>,
          <h3>
            Words: 
            unknown
</h3>,
        ],
        "className": "card-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "unknown",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "unknown",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Titles: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Words: ",
              "unknown",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Words: ",
            "unknown",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
